lalsimulation/lib/LALSimIMRPhenomTHM.c

```diff
@@ -49,6 +49,13 @@
 #include "LALSimIMRPhenomTHM_internals.c"
 #include "LALSimIMRPhenomTHM.h"
 
+typedef struct {
+    REAL8Sequence *TimeArray;  // Pointer to store TimeArray globally
+} GlobalData;
+
+// Step 2: Define the global instance of the structure
+GlobalData globalData;
```

Comment:
why is this needed in the first place?

Reply:
also - why not create a separate `LALSimIMRPhenomTHM_Modes_neha` as well, so the original remains unmodified? The value of having both functions is that you will have the original PhenomT and your version of it simultaneously available.

Reply:
...and you can add the `REAL8Sequence` that you are storing as `globalData.TimeArray` as an input to `LALSimIMRPhenomTHM_Modes_neha`, which will get rid of the need to use this global struct.